django/db/models/fields/related.py

```diff
@@ -495,6 +495,12 @@ def __init__(self, to, on_delete, from_fields, to_fields, rel=None, related_name
         self.to_fields = to_fields
         self.swappable = swappable
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        for cached_prop in ('path_infos', 'reverse_path_infos'):
+            state.pop(cached_prop, None)
+        return state
+
```

Comment:
This isn't strictly necessary; no tests failed without it being here, but it would seem prudent to perhaps do something like this for all the touched areas, for consistency and to guard against there being some way in which an unpickleable value ends up in there.